### DISCUSSION TITLE
Deprecate running Metals server on JDK8

### DISCUSSION THREAD

**Discussion by kpodsiad** [scalameta/metals#3784] [General]
I can elaborate on this in many lines, but I'll limit myself to just one - it's 2022 and it's time to do this.

cc: @tgodzik @dos65 @ckipp01 @tanishiking

**Comment by ckipp01** [comment]
💯 I'm totally on board with this.

**Comment by Arthurm1** [comment]
It would mean we could update the eclipse libraries.  I'm assuming you're going for >= JDK11?

**Reply by kpodsiad** [reply]
Yes, requiring `>= JDK11` seems reasonable.

**Reply by kpodsiad** [reply]
What I'm going to say might be a bit controversial but I think we should aim for shipping Metals with its own JDK. There is a whole category of bags that are related to the picking of Java version (I'm even facing one with inferring JDK sources now). Shipping Metals with their own JDK eliminates this category.

**Reply by Arthurm1** [reply]
Will that help with sources - as the user would still specify their own jdk for compiling wouldn't they?  I thought this would just mean that Metals could take advantage of newer JDK features, use libraries that require >= 11 and drop the jdk 8 CI build

**Reply by kpodsiad** [reply]
>Will that help with sources - as the user would still specify their own jdk for compiling wouldn't they?

It was just an example of what kind of problems we are facing now and what can be fixed with shipping with our own Java.

>I thought this would just mean that Metals could take advantage of newer JDK features, use libraries that require >= 11 and drop the jdk 8 CI build

and you thought correctly ;) There are exactly the gains from using >=jdk11.

**Comment by dos65** [comment]
@kpodsiad do you mean adding a deprecation warning? If yes then I'm for it.

**Reply by tanishiking** [reply]
I agree with @dos65, I believe we need a notification period before dropping Java8 support (I'm not sure how long we should have) as I know a lot of companies still using Java8 for some reasons, unfortunately.

**Reply by kpodsiad** [reply]
Yes, I meant adding a deprecation warning @dos65

**Reply by kpodsiad** [reply]
@tanishiking It's true that some companies still use Java 8, but my proposal concerns only Java used for running Metals server. One will be able to specify java in build tool run that code on whatever JVM they want.

**Reply by dos65** [reply]
@kpodsiad Cool, then let's do that. I'm think that having this warning for 2-3 months should be enough to inform users who are still using only jdk8.

**Reply by tanishiking** [reply]
> Yes, I meant adding a deprecation warning

Great, then it's all fine :)

**Comment by tgodzik** [comment]
One important thing to keep in mind is to make sure that when exporting Bloop or running sbt build server, we need to run the correct Java version.
